src/api/spec/features/webui/search_spec.rb

```diff
@@ -0,0 +1,332 @@
+require "sphinx_helper"
+# WARNING: If you change tests make sure you uncomment this line
+# and start a test backend. The search real backend answers for projects/packages.
+# CONFIG['global_write_through'] = true
+
+RSpec.feature "Search", type: :feature, js: true do
+  let(:admin_user) { create(:admin_user) }
+  let(:user) { create(:confirmed_user, login: 'titan' ) }
+  let(:tom) { create(:confirmed_user, login: 'tom') }
+  let(:henne) { create(:confirmed_user, login: 'henne') }
+
+  let(:package) { create(:package, name: 'goal', title: 'First goal', project_id: user.home_project.id) }
+  let(:another_package) { create(:package, name: 'goal2', title: 'Second goal', project_id: user.home_project.id) }
+  let(:bs_request) { create(:bs_request, description: "This request's description", creator: user.login) }
+
+  let(:apache) { create(:project, name: 'Apache', title: 'Awesome project', description: 'Very awesome project') }
+  let(:apache2) { create(:project, name: "#{apache.name}:Apache2", title: 'New and awesome project', description: 'Very very very awesome project') }
+  let(:apache2_subproject) { create(:project, name: "#{apache2.name}:FakeSubproject") }
+
+  let(:russian_project) { create(:project, name: 'Russian', title: 'Этёам вокябюч еюж эи') }
+  let(:chinese_project) { create(:project, name: 'Chinese', title: '窞綆腤 埱娵徖 渮湸湤 殠 唲堔') }
+
+  let(:hidden_project) { create(:forbidden_project, name: 'SecretProject', title: 'Fake description') }
+  let(:hidden_package) { create(:package, name: 'hidden_package', title: 'Hidden package rocks!', project_id: hidden_project.id) }
+
+  scenario "search link is set properly at footer" do
+    visit root_path
+
+    find('div#footer a.search-link').click
+
+    expect(page).to have_text("Search")
+    expect(page).to have_text("Advanced")
+    expect(page).to have_current_path(search_path, only_path: true)
+  end
+
+  scenario "basic search functionality" do
+    package
+    reindex_for_search
+
+    visit search_path
+    expect(page).to have_current_path(search_path, only_path: true)
+
+    fill_in 'search_input', with: package.name
+    click_button 'search_button'
+
+    expect(page).to have_current_path(search_path, only_path: true)
```

Comment:
All the `have_`**whatever** methods as DSL methods... included the `have_current_path`, but I prefer to check the path to know that I'm in the right page... instead of checking a content that can be in another view too...